code/game/machinery/computer/law.dm

```diff
@@ -31,12 +32,18 @@
 			if(!current)//no AI selected
 				to_chat(user, "<span class='danger'>No AI selected. Please chose a target before proceeding with upload.")
 				return
+			if(upload_cooldown)
+				to_chat(user, "<span class='danger'>Please wait a little before using the upload again.")
+				return
```

Comment:
Instead of making this a flag, try something like:

    if(next_upload > world.time)
        to_chat(user, "<span class='danger'>No tensor processing units are available for neural network retraining. Please try again later.</span>")
        return